experimental/list_formatter/src/list_formatter.rs

```diff
@@ -49,65 +49,75 @@ impl ListFormatter {
     fn format_internal<'c, B>(
         &'c self,
         values: &[&str],
-        empty: fn() -> B,
-        single: fn(&str) -> B,
-        apply_pattern: fn(&str, &PatternParts<'c>, B) -> B,
+        // Empty builder
+        mut builder: B,
+        // Produces value + builder
```

Comment:
nit: might be good to talk about what it should be _doing_, seems like it's more that this function consumes a value and a builder, applying it to the builder, and returns the modified builder

Reply:
This will disappear in the next PR which I'm sending now, so whatever.